packages/ui-causal-graph-editor/src/shared/rendering/engine.tsx

```diff
@@ -1860,6 +1860,13 @@ export class Engine extends PIXI.EventEmitter<EngineEvents> {
             // ensure the callback is invoked - the layout might be custom/not go through the worker
             // which fires events so just in case fire one here
             this.onLayoutComputationDoneBound();
+
+            // Force re-render to ensure nodes are positioned correctly within the frame
+            // after a new node is added or the graph is resized. This prevents nodes
+            // from being culled prematurely by ensuring the latest frame is used for culling.
+            setTimeout(() => {
+                this.viewport.dirty = true;
+            }, 300);
```

Comment:
Have you tested if the timeout needs to be this large? It's quite visible so I'm wondering if we can get away with a shorter one.

Make sure it works for larger graphs too though in case it takes a bit longer to render.

Reply:
Managed to reduce to 100 and seems to work fine, tried with differently sized graphs

Reply:
Nice, should be much less noticable